.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,158 @@
+      - uses: bahmutov/npm-install@v1
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - run: npm run build --if-present
+      - run: npm run docs:build
+      - uses: actions/upload-artifact@v2
+        with:
+          name: docs-build
+          path: |
+            docs
+            lib
+          retention-days: 1
+
+  pa11y:
+    needs: build
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      # Cache and install npm dependencies using https://github.com/bahmutov/npm-install
+      - uses: bahmutov/npm-install@v1
+      - uses: actions/download-artifact@v2
+        with:
+          name: docs-build
+      # Explicitly install testing dependencies
+      -  run: npm ci
+        working-directory: ./testing
+      - run: ruby -run -e httpd -- docs -p 6006 &
+      - run: npx pa11y-ci --config=pa11yci.local.js
+        working-directory: ./testing/wcag
+
+  backstop:
+    needs: build
+    runs-on: ubuntu-20.04
+    services:
+      backstop:
+        image: backstopjs/backstopjs:5.1.0
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v2
+        with:
+          name: docs-build
+      - run: ruby -run -e httpd -- docs -p 6006 &
+      - run: ./bin/wait-for localhost:6006 -- docker run --rm -v $(pwd)/testing/visual-regression:/src backstopjs/backstopjs test --config=backstop-config-actions.js
+      - uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: backstop-report
+          path: |
+            testing/visual-regression/backstop_data/html_report
+            testing/visual-regression/backstop_data/bitmaps_reference
+            testing/visual-regression/backstop_data/bitmaps_test
+          retention-days: 2
+
+  # The Ubuntu 20.04 virtual environment includes a copy of Chrome by default
+  # We can use this to run our selenium tests locally without needing selenium grid
+  # See: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#browsers-and-drivers
+  selenium-tests:
+    needs: build
+    runs-on: ubuntu-20.04
+    env:
+      HEADLESS: true
+      APP_ENVIRONMENT: local
+      BROWSER: chrome
+      LOG_LOCATION: automation_logs.log
+      TESTING_BASE_URL: 'http://localhost:6006'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - uses: actions/download-artifact@v2
+        with:
+          name: docs-build
+      - run: ruby -run -e httpd -- docs -p 6006 &
+      - run: ./bin/wait-for localhost:6006 -- bundle exec rake design_system:all
+      - uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: grid-artifacts
+          path: testing/artifacts
+          retention-days: 7
+
+  browserstack:
```

Comment:
This doesn't actually _run_ browserstack tests yet, but illustrates how we'd use a matrix to do that. I think in reality we might want to run this as a separate workflow file which listens to the `page_build` event as that's what we run these tests against. Running against every master build doesn't tell us anything, only when the pages build changes.

https://docs.github.com/en/actions/reference/events-that-trigger-workflows#page_build